bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureSession.java

```diff
@@ -0,0 +1,1018 @@
+                }
+                VerisureClimatesDTO cThing = new VerisureClimatesDTO();
+                VerisureClimatesDTO.Installation inst = new VerisureClimatesDTO.Installation();
+                List<VerisureClimatesDTO.Climate> list = new ArrayList<>();
+                list.add(climate);
+                inst.setClimates(list);
+                VerisureClimatesDTO.Data data = new VerisureClimatesDTO.Data();
+                data.setInstallation(inst);
+                cThing.setData(data);
+                // Set unique deviceID
+                String deviceId = climate.getDevice().getDeviceLabel();
+                if (deviceId != null) {
+                    // Set location
+                    cThing.setLocation(climate.getDevice().getArea());
+                    notifyListenersIfChanged(cThing, installation, deviceId);
+                }
+            });
+        } else {
+            logger.debug("Failed to update ClimateStatus, thing is null");
+        }
+    }
+
+    private synchronized void updateDoorWindowStatus(Class<? extends VerisureThingDTO> jsonClass,
+            VerisureInstallation installation) {
+        BigDecimal installationId = installation.getInstallationId();
+        String url = START_GRAPHQL;
+        String operation = "DoorWindow";
+        VariablesDTO variables = new VariablesDTO();
+        variables.setGiid(installationId.toString());
+        String query = "query " + operation
+                + "($giid: String!) {\n installation(giid: $giid) {\n doorWindows {\n device {\n deviceLabel\n area\n __typename\n }\n type\n state\n wired\n reportTime\n __typename\n }\n __typename\n}\n}\n";
+
+        String queryQLDoorWindow = createOperationJSON(operation, variables, query);
+        logger.debug("Quering API for door&window status");
+
+        VerisureDoorWindowsDTO thing = (VerisureDoorWindowsDTO) postJSONVerisureAPI(url, queryQLDoorWindow, jsonClass);
+        logger.debug("REST Response ({})", thing);
+
+        if (thing != null) {
+            List<VerisureDoorWindowsDTO.DoorWindow> doorWindowList = thing.getData().getInstallation().getDoorWindows();
+            doorWindowList.forEach(doorWindow -> {
+                VerisureDoorWindowsDTO dThing = new VerisureDoorWindowsDTO();
+                VerisureDoorWindowsDTO.Installation inst = new VerisureDoorWindowsDTO.Installation();
+                List<VerisureDoorWindowsDTO.DoorWindow> list = new ArrayList<>();
+                list.add(doorWindow);
+                inst.setDoorWindows(list);
+                VerisureDoorWindowsDTO.Data data = new VerisureDoorWindowsDTO.Data();
+                data.setInstallation(inst);
+                dThing.setData(data);
+                // Set unique deviceID
+                String deviceId = doorWindow.getDevice().getDeviceLabel();
+                if (deviceId != null) {
+                    // Set location
+                    dThing.setLocation(doorWindow.getDevice().getArea());
```

Comment:
Not really sure what you mean, can you give an example?

Reply:
```
public interface GiveThisABetterName{
    public Device getDevice();
}

public static class Climate implements GiveThisABetterName{
...
}
```

To make this work you need pick a `Device` class. I suggest using the one included in VerisureBaseThingDTO. The only reason that I see that you made a custom one for `Climate` and some other ones is because the `Gui` class contains addition fields. You can just merge all those extra fields into the `Gui` class in `VerisureBaseThingDTO` and everything should work the same.

Reply:
I've moved most of the DTO classes to VerisureBaseThingDTO and added all needed fields to them.